### ISSUE TITLE
Missing spaces in extract_text() method

### ISSUE THREAD

**Issue by Sunguru** [py-pdf/pypdf#1328]
Missing spaces in extract_text() method. 
See attached PDFs. 
Text is being extracted nice, but it comes with no spaces from almost all fields.

## Environment

```bash
$ python -c "import pypdf;print(pypdf.__version__)"
pypdf==3.14.0
```

## Code + PDF

PDF: [0004.pdf](https://github.com/py-pdf/PyPDF2/files/9498481/0004.pdf)

```python
from pypdf import PdfReader, __version__

print(f"pypdf=={__version__}")

reader = PdfReader("0004.pdf")

page = reader.pages[0]
extracted = page.extract_text().split("Description:")[1].split("8/11/22")[0]
print(extracted)
```

gives:

```bash
 Reportingcrudeoilleak.
Leakwasisolatedtowell
pad.Segmentoflinewas
immediatelyisolated,now
estimatedat5barrelsofoil
spilt.Rootcausestill
unknownatthistime.
```

expected (copy-pasted with Google chrome):

```
Reporting crude oil leak.
Leak was isolated to well
pad. Segment of line was
immediately isolated, now
estimated at 5 barrels of oil
spilt. Root cause still
unknown at this time.
```

[0000.pdf](https://github.com/py-pdf/PyPDF2/files/9498491/0000.pdf)

Yes, you may add to the tests. It is public `data` from here: https://northdakota.hazconnect.com/ListIncidentPublic.aspx

p,s, Thank you for the great package!

**Comment by tpcgold** [comment]
any workaround on this so far? 
I ran into the exact same issue with pypdf

**Comment by ssjkamei** [comment]
Hi.

The problem seemed to be the difference between the font size retrieved value and the actual space.
As far as the area concerned, the PDF spaces were -277.75 apart, whereas the size retrieved from the font was 278.0.
The larger font size was also -277.75 away in the area I checked, but the size I got from the font was 361.0.

If I round up the actual values, I think it will work.
However, I am not familiar with how PDFs and fonts work and I cannot determine if that is the correct process.

Is this helpful?

Try rounding up `abs(float(op)`.

https://github.com/py-pdf/pypdf/blob/8dd9fcb8d0ed06fa5230bd9a5ce5ffea80d04245/pypdf/_page.py#L1992
